packages/fast-usdc/test/cli/transfer.test.ts

```diff
@@ -0,0 +1,169 @@
+      },
+    ],
+    {
+      amount: [
+        {
+          amount: '20000',
+          denom: 'uusdc',
+        },
+      ],
+      gas: '200000',
+    },
+    'Register Account Transaction',
+  ]);
+});
+
+const makeMockEthProvider = () => {
+  const txnArgs: any[] = [];
+  const provider = {
+    getTransactionCount: () => {},
+    estimateGas: () => {},
+    getNetwork: () => ({ chainId: 123 }),
+    getFeeData: () => ({ gasPrice: 1 }),
+    broadcastTransaction: (...args) => {
+      txnArgs.push(args);
+      return {
+        blockNumber: 9000,
+      };
+    },
+    getTransactionReceipt: () => ({
+      blockNumber: 9000,
+      hash: 'SUCCESSHASH',
+      confirmations: () => [9000],
+      logs: [],
+    }),
+  };
+
+  return { provider, getTxnArgs: () => txnArgs };
+};
+
+test('Transfer signs and broadcasts the depositForBurn message on Ethereum', async t => {
+  const path = 'config/dir/.fast-usdc/config.json';
+  const nobleApi = 'http://api.noble.test';
+  const nobleToAgoricChannel = 'channel-test-7';
+  const config = {
+    agoricRpc: 'http://rpc.agoric.test',
+    nobleApi,
+    nobleToAgoricChannel,
+    nobleSeed: 'test noble seed',
+    ethRpc: 'http://rpc.eth.test',
+    ethSeed: 'a4b7f431465df5dc1458cd8a9be10c42da8e3729e3ce53f18814f48ae2a98a08',
+    tokenMessengerAddress: '0x9f3B8679c73C2Fef8b59B4f3444d4e156fb70AA5',
+    tokenAddress: '0x1c7D4B196Cb0C7B01d743Fbc6116a902379C7238',
+  };
+  const out = mockOut();
+  const file = mockFile(path, JSON.stringify(config));
+  const agoricSettlementAccount = 'agoric123456';
+  const settlementAccountVstoragePath = 'published.fastUSDC.settlementAccount';
+  const vstorageMock = makeVstorageMock({
+    [settlementAccountVstoragePath]: agoricSettlementAccount,
+  });
+  const amount = '150';
+  const destination = 'dydx1234';
+  const nobleFwdAccountQuery = `${nobleApi}/noble/forwarding/v1/address/${nobleToAgoricChannel}/${agoricSettlementAccount}+${destination}`;
+  const fetchMock = makeFetchMock({
+    [nobleFwdAccountQuery]: {
+      address: 'noble14lwerrcfzkzrv626w49pkzgna4dtga8c5x479h',
+      exists: true,
+    },
+  });
+  const nobleSignerAddress = 'noble09876';
+  const signerMock = makeMockSigner();
+  const mockEthProvider = makeMockEthProvider();
+
+  await transfer.transfer(
+    file,
+    amount,
+    destination,
+    // @ts-expect-error mocking console
+    out,
+    fetchMock.fetch,
+    vstorageMock.vstorage,
+    { signer: signerMock.signer, address: nobleSignerAddress },
+    mockEthProvider.provider,
+  );
+
+  t.is(signerMock.getSigned(), undefined);
+  t.deepEqual(mockEthProvider.getTxnArgs()[0], [
+    '0xf8a4800180941c7d4b196cb0c7b01d743fbc6116a902379c723880b844095ea7b30000000000000000000000009f3b8679c73c2fef8b59b4f3444d4e156fb70aa50000000000000000000000000000000000000000000000000000000008f0d18082011aa0b2d87eeb1cb36243f95662739e2a7bd4bddc2b8afe189ac4848ec71cc314335ba068136695c644f69474e2e30ea7059f9b380fbb1a09beb3580f73d3ea349912ab',
+  ]);
```

Comment:
I wonder how to review that. I suppose we'll eventually do integration testing, and we'll find out real quick if it's not right at that point.

Meanwhile, I did try to puzzle it out a little, and this form might be slightly more legible. Dunno whether it's really useful, but I'll share it rather than throw it away:

```suggestion
  t.deepEqual(ethers.decodeRlp(mockEthProvider.getTxnArgs()[0]), [
    '0x',
    '0x01',
    '0x',
    '0x1c7d4b196cb0c7b01d743fbc6116a902379c7238',
    '0x',
    '0x095ea7b30000000000000000000000009f3b8679c73c2fef8b59b4f3444d4e156fb70aa50000000000000000000000000000000000000000000000000000000008f0d180',
    '0x011a',
    '0xb2d87eeb1cb36243f95662739e2a7bd4bddc2b8afe189ac4848ec71cc314335b',
    '0x68136695c644f69474e2e30ea7059f9b380fbb1a09beb3580f73d3ea349912ab',
  ]);
```

Reply:
Yea I'm not really sure either. Just left a comment because I found a website that can decode it. But I spent a little bit trying to do it myself with `ethers` and couldn't succeed.

Reply:
cool. the comment is quite nice.